Content.Shared/Vehicle/SharedVehicleSystem.cs

```diff
@@ -129,26 +131,71 @@ private void OnVehicleStartup(EntityUid uid, VehicleComponent component, Compone
     /// change its draw depth. Vehicles can choose between special drawdetph
     /// when facing north or south. East and west are easy.
     /// </summary>
-    protected int GetDrawDepth(TransformComponent xform, bool northOnly)
+    protected int GetDrawDepth(TransformComponent xform, VehicleComponent component, double cameraDegrees)
     {
-        // TODO: I can't even
-        if (northOnly)
+        var cameraAngle = GetAngle(cameraDegrees + 90);
+
+        var itemAngle = GetAngle(xform.LocalRotation.Degrees + cameraAngle + 90);
+
+        var itemDirection = string.Empty;
+        var cameraDirection = cameraAngle switch
         {
-            return xform.LocalRotation.Degrees switch
-            {
-                < 135f => (int) DrawDepth.DrawDepth.Doors,
-                <= 225f => (int) DrawDepth.DrawDepth.WallMountedItems,
-                _ => 5
-            };
+            >= 0 and < 90 => "Right",
+            >= 90 and < 180 => "Down",
+            >= 180 and < 270 => "Left",
+            _ => "Up"
+        };
```

Comment:
use the `Direction` enum instead of strings like this

Reply:
I did it.